### PR TITLE
Updates to allow for testing seeder without writing to data folder

### PR DESCRIPTION
**Pull request by dsimmons87**
Adds the following:

**Dependency Injection:**

This package allows for making some defined modules and classes to be hot-swappable, which will aid with testing and upgrading, and also help make the code more modular.

It also allows for function arguments to be automatically pulled from its containers, but this doesn't work in python 3.5, so that will need to wait until python is updated.

*Edit:* As the server is now running python 3.7, I've change this now to inject the dependencies in the couple of places that the filesystem is used so far.

**FS:**

The FS module has several different implementations of a base FS structure, which makes it easy to swap out the OS file system for one in memory for testing for example. It also allows for Zips to be accessed and treated almost the same way as other parts of the file system so that may come in useful later.

**Updates to process_upload in handlers.php:**

I've swapped out some of the direct file system operations with some which use the FS module served from the service container. This allowed for the FS to be hot-swapped in the seeder test.

For now I've just manually tested it on the RC, but I'm hoping to break this code out into a more testable structure later.

### CODE REVIEW

File: openra/containers.py [hunk 1]
```diff
@@ -4,10 +4,10 @@
 from openra import settings
 from os import path
 
-class ServiceContainer(containers.DeclarativeContainer):
+class Container(containers.DeclarativeContainer):
     config = providers.Configuration()
 
-    fs = providers.Singleton(
+    dataFs = providers.Singleton(
```
Comment (pchote) [inline openra/containers.py:10]: Naming nit: `data_fs`?

File: openra/tests/test_commands.py [hunk 2]
```diff
@@ -18,17 +21,17 @@ def runSeeder(self):
...
-        container.fs = providers.Singleton(
+        container.dataFs.override(providers.Singleton(
             MemoryFS
-        )
+        ))
 
-    def oramapFileExistsForMapId(self, mapId):
-        fs = container.fs()
+    @inject
+    def oramapFileExistsForMapId(self, mapId, dataFs:FS=Provide[Container.dataFs]):
```
Comment (pchote) [inline openra/tests/test_commands.py:29]: Likewise

**Review by pchote — COMMENTED** [review]
LGTM aside from a couple of small naming nits.

Not tested, so i'm trusting you have it under control!

File: openra/containers.py (hunk 1 above)
Comment (dsimmons87) [reply openra/containers.py]: Ok this is now snake_case

**Pull request merged**